[WiP]: Add `data-list-item-id` attribute on `<li>` elements in data downcast (#18513)
ckeditor/ckeditor5: packages/ckeditor5-list/docs/features/lists.md

	The {@link module:list/listproperties~ListProperties} feature overrides UI button implementations from the {@link module:list/list/listui~ListUI}.
</info-box>

### Editor output configuration

> Starting with version 45.1.1, list items (`<li>` elements) in the editor data output contain an additional attribute called `data-list-item-id`. It's necessary to ensure better data comparisons within some editor features, like revision history.

Reviewer: ```suggestion
Starting with version 45.1.1, list items (`<li>` elements) in the editor data output contain an additional attribute `data-list-item-id`. The attribute is necessary to ensure that the lists feature work correctly with other editor features and mechanisms.
```

```html
<ul>
	<li data-list-item-id="e72808ee4144975064acb5d66e5cfba13">Hello</li>
	<li data-list-item-id="e5d719ab356409767e9d4358485476358">There!</li>
</ul>
```

> If data loaded to the editor does not contain the `data-list-item-id` attribute on list items, it will be automatically added when data is saved.

Reviewer: ```suggestion
If data loaded in the editor does not contain the `data-list-item-id` attributes on list items, they will be automatically added when data is saved.
```

> If you need to save the editor contents without this attribute, you can set the `skipListItemIds` flag to `true` when calling `editor.getData()` method:

Reviewer: ```suggestion
If you wish to save the editor contents without this attribute, you can set the `skipListItemIds` flag to `true` when calling `editor.getData()` method. **However, please note, that data without IDs should be used only for presentation purposes (e.g. displaying the document as HTML), and should not be used to initialize the editor.**
```

```js
editor.getData( { skipListItemIds: true } );
```

Such output will not contain `data-list-item-id` attribute:

```html
<ul>
	<li>Hello</li>
	<li>There!</li>
</ul>
```

<info-box>
	It is not recommended to use data without `data-list-item-id` attribute in the long run as some features may work incorrectly.

> </info-box>

Reviewer: I wouldn't even suggest that it is okay to use this `data-list-item-id` as "editor data".

I proposed a different message in comment above. We could have it in a box as well (warning box?), but I'd like to have this warning box directly after the paragraph that mentions `skipListItemIds` so there's a bigger chance that people will read it.

Author: I think putting it in bold as you proposed will be enough.
